Assets/Scripts/Game/PlayerAmbientLight.cs

```diff
@@ -52,34 +56,49 @@ void Update()
                 StartCoroutine(ChangeAmbientLight());
         }
 
+        bool mutex;
```

Comment:
To ensure that "mutex" changes are immediately seen by all threads, I think it should at very least be declared volatile.
Busy looping is also terrible. If you don't want to go the full Mutex class way, use at least a lock() statement